indico/modules/events/management/controllers/protection.py

```diff
@@ -63,42 +70,86 @@ def _process(self):
                                 endpoint='event_management.acl')
 
 
-class RHEventProtection(RHManageEventBase):
+class RHEventProtectionBase(RHManageEventBase):
+    def __init__(self):
+        super(RHEventProtectionBase, self).__init__()
+        selectable_permissions = {k: v for k, v in get_available_permissions(Event).viewitems() if v.user_selectable}
+        self.whitelisted_permissions = set(selectable_permissions.keys()) | {'edit', 'access'}
+        self.permissions_tree = {
+            'edit': {
+                'title': _('Event Management'),
+                'children': {
+                    v.name: {'title': v.friendly_name} for k, v in selectable_permissions.viewitems()
+                }
+            }
+        }
+        self.permissions_tree['edit']['children']['access'] = {'title': _('Access')}
+        self.permissions_tree['edit']['children'] = OrderedDict(
+            sorted(self.permissions_tree['edit']['children'].items(), key=lambda t: t[0])
+        )
+
+
+class RHEventProtection(RHEventProtectionBase):
     """Show event protection"""
 
     NOT_SANITIZED_FIELDS = {'access_key'}
 
     def _process(self):
         form = EventProtectionForm(obj=FormDefaults(**self._get_defaults()), event=self.event)
         if form.validate_on_submit():
+            permission_principals = defaultdict(set)
+            for principal_data in form.permissions.data:
```

Comment:
`for what, ever in form.permissions.data` - like this you have meaningful names instead of `principal_data[0]` and `principal_data[1]`